File: static_media/code/birds-on-a-wire.py
l,d,n = map(int, input().split())
birds = [int(input()) for _ in range(n)]
birds.sort()
offset = 6

def fill_interval(start, end, gap):
    return (end - start) // gap + 1 if start <= end else 0

if n == 0:
    print(fill_interval(0, l - 2*offset, d))
else:
    total = 0
    for i in range(n+1):
        start = offset if i==0 else birds[i-1] + d
        end = l - offset if i==n else birds[i] - d
        total += fill_interval(start, end, d)
        
    print(total)
    

File: static_media/code/birds-on-a-wire.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;
using ll = long long;

// birds can't sit any closer to the poles than this
const ll offset = 6;

int main() {
  ll n, l, d;
  cin >> l >> d >> n;
  vector<ll> birds(n);
  for (auto &b : birds)
    cin >> b;
  sort(birds.begin(), birds.end());

  if (n == 0) {
    cout << (l - 2 * offset) / d + 1 << '\n';
    return;
  }

  int totalBirds = 0;
  for (int i = 0; i <= n; i++) {
    ll start = i == 0 ? offset : birds[i - 1] + d;
    ll end = i == n ? l - offset : birds[i] - d;
    if (start <= end) {
      totalBirds += (end - start) / d + 1;
    }
  }
  cout << totalBirds << '\n';
  return 0;
}